applications/plugins/BeamPlastic/forceField/BeamPlasticFEMForceField.inl

```diff
@@ -0,0 +1,2530 @@
+    , d_ySection(initData(&d_ySection, (Real)0.2, "ySection", "length of the section in the y direction for rectangular beams"))
+    , d_useSymmetricAssembly(initData(&d_useSymmetricAssembly,false,"useSymmetricAssembly","use symmetric assembly of the matrix K"))
+    , d_isTimoshenko(initData(&d_isTimoshenko,false,"isTimoshenko","implements a Timoshenko beam model"))
+    , m_edgeHandler(NULL)
+{
+    m_edgeHandler = new BeamFFEdgeHandler(this, &m_beamsData);
+
+    d_poissonRatio.setRequired(true);
+    d_youngModulus.setReadOnly(true);
+}
+
+template<class DataTypes>
+BeamPlasticFEMForceField<DataTypes>::BeamPlasticFEMForceField(Real poissonRatio, Real youngModulus, Real yieldStress, Real zSection,
+                                                    Real ySection, bool useVD, bool isPlasticMuller, bool isTimoshenko,
+                                                    bool isPlasticKrabbenhoft, bool isPerfectlyPlastic,
+                                                    helper::vector<defaulttype::Quat> localOrientations)
+    : m_beamsData(initData(&m_beamsData, "beamsData", "Internal element data"))
+    , m_indexedElements(NULL)
+    , d_poissonRatio(initData(&d_poissonRatio,(Real)poissonRatio,"poissonRatio","Potion Ratio"))
+    , d_youngModulus(initData(&d_youngModulus,(Real)youngModulus,"youngModulus","Young Modulus"))
+    , d_yieldStress(initData(&d_yieldStress, (Real)yieldStress, "yieldStress", "yield stress"))
+    , d_usePrecomputedStiffness(initData(&d_usePrecomputedStiffness, true, "usePrecomputedStiffness",
+                                        "indicates if a precomputed elastic stiffness matrix is used, instead of being computed by reduced integration"))
+    , d_useConsistentTangentOperator(initData(&d_useConsistentTangentOperator, false, "useConsistentTangentOperator", 
+                                             "indicates wether to use a consistent tangent operator in the computation of the plastic stiffness matrix"))
+    , d_isPerfectlyPlastic(initData(&d_isPerfectlyPlastic, false, "isPerfectlyPlastic", "indicates wether the behaviour model is perfectly plastic"))
+    , d_modelName(initData(&d_modelName, std::string("RambergOsgood"), "modelName", "the name of the 1D contitutive law model to be used in plastic deformation"))
+    , d_zSection(initData(&d_zSection, (Real)zSection, "zSection", "length of the section in the z direction for rectangular beams"))
+    , d_ySection(initData(&d_ySection, (Real)ySection, "ySection", "length of the section in the y direction for rectangular beams"))
+    , d_useSymmetricAssembly(initData(&d_useSymmetricAssembly,false,"useSymmetricAssembly","use symmetric assembly of the matrix K"))
+    , d_isTimoshenko(initData(&d_isTimoshenko, isTimoshenko, "isTimoshenko", "implements a Timoshenko beam model"))
+    , m_edgeHandler(NULL)
+{
+    m_edgeHandler = new BeamFFEdgeHandler(this, &m_beamsData);
```

Comment:
I reckon it would be best to use a std::unique_ptr instead of a raw pointer. If you do that, you can get rid of the destructor implementation